plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/TestSyntheticColumnHandleBuilder.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.jdbc;
+
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import static io.trino.plugin.jdbc.TestingJdbcTypeHandle.JDBC_VARCHAR;
+import static io.trino.spi.type.VarcharType.VARCHAR;
+import static java.lang.Integer.MAX_VALUE;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestSyntheticColumnHandleBuilder
+{
+    private final SyntheticColumnHandleBuilder builder = new SyntheticColumnHandleBuilder();
+
+    @DataProvider(name = "columns")
+    public static Object[][] primeNumbers()
+    {
+        return new Object[][] {
+                {"column_0", 999, "column_0_999"},
+                {"column_with_over_twenty_characters", 100, "column_with_over_twenty_ch_100"},
+                {"column_with_over_twenty_characters", MAX_VALUE, "column_with_over_tw_2147483647"}
+        };
+    }
+
+    @Test(dataProvider = "columns")
+    public void testColumnAliasTruncation(String columnName, int nextSynthenticId, String expectedSyntheticColumnName)
+    {
+        JdbcColumnHandle column = getDefaultColumnHandleBuilder()
+                .setColumnName(columnName)
+                .build();
+
+        JdbcColumnHandle result = builder.get(column, nextSynthenticId);
+
+        assertThat(result.getColumnName()).isEqualTo(expectedSyntheticColumnName);
+    }
+
+    @Test(expectedExceptions = IllegalArgumentException.class)
```

Comment:
we avoid expectedExceptions on non-oneliner tests since it's unclear which line is expected to throw this. See also https://github.com/trinodb/trino/issues/6974.

Reply:
So how do you want me to change this? try/catch?

Reply:
There are examples in the PRs linked to the issue I shared, please spend some time looking at some of those to get an idea of what options exist.

In most cases it's `assertThatThrownBy` but in some cases there are also other ways like `assertQueryFailed` or `try...catch`.